src/main/scala/com/cognite/spark/datasource/CdpRelation.scala

```diff
@@ -74,12 +112,14 @@ abstract class CdpRelation[T <: Product: Decoder](config: RelationConfig, shortN
     )
   }
 
-  def urlsWithFilters(filters: Array[Filter], uri: Uri): Seq[Uri] =
-    fieldsWithPushdownFilter
-      .map(f => (f, filters.flatMap(getFilter(_, f)))) // map from fieldname to its filtering values
-      .flatMap(pdf => pdf._2.map(v => PushdownFilter(pdf._1, v)))
-      .distinct
-      .map(f => uri.param(f.fieldName, f.value))
+  def urlsWithFilters(filters: Array[Filter], uri: Uri): Seq[Uri] = {
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val getAll = shouldGetAll(pushdownFilterExpression)
+    val params = pushdownToParameters(pushdownFilterExpression)
+    val urlsWithFilter = pushdownToUri(params, uri).distinct
+
+    if (urlsWithFilter.isEmpty || getAll) { Seq(uri) } else urlsWithFilter
```

Comment:
I guess we've been getting sloppy about this but we really should do this with full braces and newlines.
```
if (...) {
  ...
} else {
  ...
}